src/pudl/transform/ferc1.py

```diff
@@ -5877,133 +5921,57 @@ def add_dimension_total_calculations(
-        # we are not linking up the total parents to both the leafy subdimensions and
-        # the dimensions
-        | (
-            (calc_components.total_count_parent > 1)
-            & (calc_components[dimensions] != "total").all(axis=1)
-        )
-    ]
+    calc_components = calc_components.fillna(child_values)
+
     calc_cols = ["table_name", "xbrl_factoid"] + dimensions
     calc_and_parent_cols = calc_cols + [f"{col}_parent" for col in calc_cols]
+
     check_for_calc_components_duplicates(
         calc_components,
         table_names_known_dupes=[
-            "electric_plant_depreciation_functional_ferc1",
             "electricity_sales_by_rate_schedule_ferc1",
         ],
         idx=calc_and_parent_cols,
     )
+
+    # only drop duplicates if the table_name is in known dupes list.
     calc_components = calc_components.drop_duplicates(
         calc_and_parent_cols, keep="first"
     )
-    # we assigned True for the specific flag boolean columns for each dim. now fill
-    # everything else in as false.
-    flag_cols = calc_components.filter(like="is_within_dimension_").columns
-    calc_components.loc[:, flag_cols] = calc_components.loc[:, flag_cols].fillna(False)
     assert calc_components[calc_components.duplicated()].empty
     return calc_components
-
-
-def remove_non_totals_from_impiled_child_calculation_components(
-    calc_components: pd.DataFrame, dimensions: list[str]
-) -> pd.DataFrame:
-    """Remove the non-totals from child calcs when their parents are either null or total only.
-
-    When a parent factoid only has either a null or total value for a dimension AND
-    the child factoid has multiple values for its dimension, we want to remove all of
-    the calculation component records which do not have "total" for a value in that
-    dimension.
-
-    Args:
-        calc_components: dataframe of calculation components with child and parent
-            dimensions.
-        dimensions: list of dimension columns.
-    """
-    base_idx = ["table_name", "xbrl_factoid"]
-    base_parent_idx = [c + "_parent" for c in base_idx]
-    for dim in dimensions:
-        # ADD some helper columns for the mask
-        calc_components = add_has_totals_column_for_dim(
-            calc_components, idx=base_parent_idx + base_idx, dim=dim
-        )
-        # when a parent factoid only has either a null or a total for its dim
-        # AND the child factoid has multiple values for its dimension, we want to remove
-        # all of the non-totals
-        remove_mask = (
-            (calc_components[f"{dim}_parent"].isin([pd.NA, "total"]))
-            & (calc_components[f"{dim}_has_totals"])
-            & (calc_components[dim] != "total")
-        )
-        logger.info(f"{dim}: Removing {len(calc_components[remove_mask])}")
-        calc_components = calc_components[~remove_mask]
-    return calc_components
-
-
-def add_has_totals_column_for_dim(
```

Comment:
These cols were only used in the above method, afaict, so I removed them entirely.